src/http.rs

```diff
@@ -86,11 +83,22 @@ impl HttpClient {
             .post(&format!("{}/entity", self.uri))
             .headers(self.headers.clone())
             .body(s)
-            .send()?
-            .text()?;
-
-        let edn_resp = Edn::from_str(&resp.replace("#inst", ""));
-        edn_resp.or(Ok(edn!({:status ":internal-server-error", :code 500})))
+            .send()?;
+
+        if resp.status().as_u16() < 300 {
+            let resp_body = resp.text()?;
+            let edn_resp = Edn::from_str(&resp_body.replace("#inst", ""));
+            edn_resp.or(Err(CruxError::BadRequestError(format!(
+                "entity responded with {} for id \"{}\" ",
+                500, crux_id
```

Comment:
Also, why is this a 500? I didn't get it.

Reply:
This is legacy and just means that the process failed. not going to change this now

Reply:
Also it would be super weird to return an `Err` containing a status code lower than 300